Lib/sqlite3/dump.py

```diff
@@ -20,6 +20,19 @@ def _force_decode(bs, *args, **kwargs):
     except UnicodeDecodeError:
         return "".join([chr(c) for c in bs])
 
+class _ctx_text_factory:
+    def __init__(self, connection, text_factory):
+        self.connection = connection
+        self.text_factory = text_factory
+        self.orig_text_factory = None
+
+    def __enter__(self):
+        self.orig_text_factory = self.connection.text_factory
+        self.connection.text_factory = self.text_factory
+
+    def __exit__(self, type, value, tb):
+        self.connection.text_factory = self.orig_text_factory
```

Comment:
Sorry, I should have been more specific; you can use `contextlib.contextmanager`:

```python
from contextlib import contextmanager

@contextmanager
def text_factory(con, factory):
    saved_factory = con.text_factory
    con.text_factory = factory
    try:
        yield
    finally:
        con.text_factory = saved_factory
```

Reply:
Sorry, was not familiar with this until now, it much more convenient then writing a class.

Reply:
`contextlib.contextmanager` is one of my stdlib favourites! :)